### PR TITLE
Revise friendly stdlib

### PR DESCRIPTION
**Pull request by vchuravy**
Revise.jl requires a record of the files that are included in the sysimage.

cc: @timholy

As a note I am seeing a weird memory corruption issue.
Some of the strings get partially zeroed out: 
`"\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0Base64"`

### CODE REVIEW

File: base/sysimg.jl [hunk 1]
```diff
@@ -500,31 +500,52 @@ using Base
 # Ensure this file is also tracked
 pushfirst!(Base._included_files, (@__MODULE__, joinpath(@__DIR__, "sysimg.jl")))
 
+macro stdlib(name)
+    println(name)
```
Comment (timholy) [inline base/sysimg.jl]: Is this `println` deliberate?

**Review by timholy — COMMENTED** [review]
LGTM, I hadn't realized how easy this would be. Any particular reason for the macro or is it just for brevity?

File: base/sysimg.jl (hunk 1 above)
Comment (vtjnash) [reply base/sysimg.jl]: It looks like vchuravy intended to write `function`, but instead wrote `macro`? :P

Jokes aside though, we don't want to define anything (variables, functions, macros, etc) in this namespace.

**Comment by vtjnash** [comment]
> Some of the strings get partially zeroed out:

you can filter those out (where "those" = starts-with a null byte)

File: base/sysimg.jl (hunk 1 above)
Comment (vchuravy) [reply base/sysimg.jl]: Oh the original intend was to get a progress message. So the `println` was intended

**Comment by vchuravy** [comment]
> > Some of the strings get partially zeroed out:
> you can filter those out (where "those" = starts-with a null byte)

Yeah but they actually point to files that I want to include and nothing should just zero **parts** of a String.
I tried several attempts with various combination of `deepcopy` and all the strings are fine when they are stored in the list.

@timholy @maleadt mentioned to me that there is an alternative where we use the Git repo as a cache instead of having to maintain the cache ourselves.
See https://github.com/jrevels/Revise.jl/commit/ef37e8c53433ee0416437ffcc77d637116a7de68 for a prototype.

File: base/sysimg.jl (hunk 1 above)
Comment (KristofferC) [reply base/sysimg.jl]: Printing would be great. I hate looking at `Docs.jl`°

Comment (vtjnash) [reply base/sysimg.jl]: True. I think that once logging can be enabled from the command line, setting `env JULIA_DEBUG_LOADING=1` in the Makefile (as we could do on v0.6) would be a good way to do this in a generic way.

**Comment by vtjnash** [comment]
> Yeah but they actually point to files that I want to include and nothing should just zero parts of a String

Some of the Strings start with null, to indicate that they should be hidden from the includes list (`parse_cache_header` filters them)

**Pull request closed without merging**